azurerm/resource_arm_key_vault_secret.go

```diff
@@ -0,0 +1,240 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/dataplane/keyvault"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmKeyVaultSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKeyVaultSecretCreate,
+		Read:   resourceArmKeyVaultSecretRead,
+		Update: resourceArmKeyVaultSecretUpdate,
+		Delete: resourceArmKeyVaultSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateKeyVaultSecretName,
+			},
+
+			"vault_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"value": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"content_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmKeyVaultSecretCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).keyVaultManagementClient
+
+	log.Print("[INFO] preparing arguments for AzureRM KeyVault Secret creation.")
+
+	name := d.Get("name").(string)
+	keyVaultBaseUrl := d.Get("vault_uri").(string)
+	value := d.Get("value").(string)
+	contentType := d.Get("content_type").(string)
+	tags := d.Get("tags").(map[string]interface{})
```

Comment:
I'd agree, but afaik HCL is returning an empty map here - given we use [this pattern throughout the provider](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_resource_group.go#L93) - example from `azurerm_resource_group`:

```
$ cat main.tf
resource "azurerm_resource_group" "test"{
  name = "tom"
  location = "westus"
  tags { }
}
```

```
$ envchain azurerm terraform plan -refresh
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed. Cyan entries are data sources to be read.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

  + azurerm_resource_group.test
      location: "westus"
      name:     "tom"
      tags.%:   "<computed>"


Plan: 1 to add, 0 to change, 0 to destroy.
✔ ~/code/src/tmp/jake
17:01 $ envchain azurerm terraform apply
azurerm_resource_group.test: Creating...
  location: "" => "westus"
  name:     "" => "tom"
  tags.%:   "" => "<computed>"
azurerm_resource_group.test: Creation complete (ID: /subscriptions/xxxxxxxxxx/resourceGroups/tom)
```